### PR TITLE
Bug #502 x asl spm smooth

### LINKED ISSUE #502: xASL_spm_smooth: Fix smoothing 4D NifTI

### PR DESCRIPTION
**Pull request by HenkMutsaerts**
Closes #502

### CODE REVIEW

**Review by MichaelStritt — APPROVED** [review]
I checked out all the changes. Seems fine to me. Did you test it?

**Review by BeatrizPadrela — APPROVED** [review]
It works!

**Comment by HenkMutsaerts** [comment]
> It works!

Thanks for testing and finding the bug!

**Review by jan-petr — APPROVED** [review]
OK.

File: External/SPMmodified/xASL/xASL_spm_smooth.m [hunk 1]
```diff
@@ -117,15 +117,20 @@ function xASL_spm_smooth(pathIn, fwhmSmooth, pathNew)
...
+
 if ~isempty(pathNew) % otherwise don't rename, keep prefix "s" for smoothed
 
     TempName = fullfile(Fpath, ['s' Ffile Fext]);
-
+    
     if ~strcmp(pathNew,TempName)
-        pathDest=fullfile(Fpath, pathNew);
-        xASL_Move(TempName,pathDest,1);
+        [FpathN, ~, ~] = xASL_fileparts(pathNew);
```
Comment (HenkMutsaerts) [inline External/SPMmodified/xASL/xASL_spm_smooth.m:126]: very nice! Note that you don't need the ~ ~ (so FpathN is sufficient).

File: External/SPMmodified/xASL/xASL_spm_smooth.m [hunk 2]
```diff
@@ -117,15 +117,20 @@ function xASL_spm_smooth(pathIn, fwhmSmooth, pathNew)
...
+        if isempty(FpathN) % This means that the user didn't choose a fullpath, only a file name, so FpathN = ' '.
+            pathDest=fullfile(Fpath, pathNew);
+            xASL_Move(TempName,pathDest,1);
```
Comment (HenkMutsaerts) [inline External/SPMmodified/xASL/xASL_spm_smooth.m:129]: I would insert `xASL_adm_CreateDir(Fpath);` here to be sure. This function creates the Fpath folder only if it doesn't exist!

**Review by HenkMutsaerts — COMMENTED** [review]
Two small requests, then you can approve and merge ;)

**Pull request merged**